### ISSUE TITLE
Integration not working

### ISSUE THREAD

**Issue by chaoscrack** [siku2/hass-dingz#18]
Dear all

The integration is not longer working. I got the following logs from HomeAssistant:

'SystemConfig' unhandled keys: {'id', 'protected_status', 'dyn_light', 'sun_offset', 'sunset', 'wifi_ps', 'allow_remote_reboot', 'sunrise', 'lat', 'long', 'tzid'}
08:32:03 – (WARNUNG) dingz (benutzerdefinierte Integration) - Die Nachricht ist zum ersten Mal am 08:32:03 aufgetreten und erscheint 2 mal
'Device' unhandled keys: {'dip_misconf', 'dip_static'}
08:32:03 – (WARNUNG) dingz (benutzerdefinierte Integration)
Error setting up entry C50000 for dingz
08:32:03 – (FEHLER) dingz (benutzerdefinierte Integration)
Unable to find referenced entities light.dingz_led or it is/they are currently not available
08:19:14 – (WARNUNG) helpers/service.py - Die Nachricht ist zum ersten Mal am 08:17:12 aufgetreten und erscheint 10 mal

Thanks for helping

**Comment by epijim** [comment]
Did you update firmware to 2.0.14? I have a single switch on 2.0.14 and it doesn't work, but all the other switches on 1.3.30 are fine.

**Comment by chaoscrack** [comment]
Yes, I'm on FW 2.0.14

**Comment by mullerch** [comment]
Also have this issue. Sine update 2.0.x, the API has [breaking changes](http://dingz.ch/en/changelog-en/) and the integration is therefore no more compatible.

**Comment by mullerch** [comment]
@pail23 Hi Paul, I see you're working on it. I wanted to fix it but if you have an ETA I would be happy to wait :-)

**Comment by pail23** [comment]
@mullerch I will stop working on the fix since there are some issues with the new firmware version (e.g. after changing a light, the dingz device does not accept API calls for around 2 seconds). I moved my setup to mqtt.

**Comment by mullerch** [comment]
Thanks for the update. In any case, the MQTT implementation seems to me more relevant, especially for subscriptions where we can avoid polling. The only necessary REST call would be the configuration of the MQTT server.

**Comment by beathoch** [comment]
Hello Christian, I tried with MQTT too. My broker works perfectly and my local explorer sees the activities. But I can't put HA devices on. As I have read, the HA Explorert does not see that as a device. You probably have to create this manually in the configuration.yaml. But that's exactly where I'm gonna stop, I have no idea how I have to configure with it. Do you have a example for me? or a kickstart?

**Comment by mullerch** [comment]
I'm not an expert, but I would go with the mqtt platform : https://www.home-assistant.io/integrations/mqtt/#manual-configured-mqtt-items.

Here is an example on how this could look like:

```
mqtt:
  cover:
    - command_topic: "dingz/<ID>/<model>/command/motor"
      name: "Living room cover"
      payload_open: "{"position": 100,“goal”:50,"lamella":30, “motion”:2}"
      payload_close: "xxx"
      payload_stop: "xxx"
```

You must of course adapt for the dingz.

It's a bit cumbersome and a complete python integration will of course provide better ease of use than this.

**Comment by beathoch** [comment]
Thank you very much for the example! This helps me further, I will experiment with it and I hope to configure my light switches. If you integrate the dingz into HA via homekit, you have the devices including all sensors available. That's how i have it right now, but with MQTT I think it's the better way. And the homekit is available for Apple Home.

**Comment by OdyX** [comment]
Some parts of the mentionned issues should be fixed by v2.0.0. Please retry and report.

**Comment by siku2** [comment]
> @mullerch I will stop working on the fix since there are some issues with the new firmware version (e.g. after changing a light, the dingz device does not accept API calls for around 2 seconds). I moved my setup to mqtt.

@pail23, I'm interested to hear how that went for you. I was playing around with the MQTT API  while I was working on the rewrite for this integration. I was even planning on switching to it completely. However, in typical Dingz fashion, it seems broken.
Regardless of which output is turned on, the dingz only ever sends feedback for channel '0' (i.e. `state/light/0` and `state/motor/0`).
Did the MQTT interface work according to the specification for you?
